fix(lsp): don't treat MarkedString[] with language id as empty

rmuir opened the pull request:
Problem:
Hover response of MarkedString[] where the first element contains a language identifier treated as empty.

Solution:
Fix empty check to handle case of MarkedString[] where the first element is a pair of a language and value.

Closes #35516

MariaSolOs commented on runtime/lua/vim/lsp/buf.lua:
```diff
@@ -72,7 +72,13 @@ function M.hover(config)
         if
           (
             type(result.contents) == 'table'
```
This if condition is quite hard to follow at this point. Could we split it up into smaller/more readable blocks, or at least add a comment explaining each case that this condition checks for?

rmuir replied:
I agree. I will give it a try, not a native lua speaker, went for the most conservative fix.

rmuir replied:
Added comments to explain the 5 different datastructures at play here...

rmuir committed:
fix(lsp): don't treat MarkedString[] with language id as empty

Problem:
Hover response of MarkedString[] where the first element contains a
language identifier treated as empty.

Solution:
Fix empty check to handle case of MarkedString[] where the first element
is a pair of a language and value.

justinmk commented on runtime/lua/vim/lsp/buf.lua:
```diff
@@ -69,11 +69,26 @@ function M.hover(config)
+        -- - MarkedString-string: "doc"
+        -- - MarkedString-pair: { language="c", value="doc" }
+        -- - MarkedString[]-string: { "doc1", ... }
+        -- - MarkedString[]-pair: { { language="c", value="doc1" }, ... }
```
This is a great comment!
